### PR TITLE
dependency: copy external tap-go package as an internal package

### PR DESCRIPTION
**Pull request by fmuyassarov**
This PR fixes subset of the dependency issues described in https://github.com/opencontainers/runtime-tools/issues/758.

Move/copy external tap-go package from [github.com/mndrix/tap-go](https://github.com/mndrix/tap-go) to be in tree package. 
[github.com/mndrix/tap-go](https://github.com/mndrix/tap-go) is no longer maintained and it is currently archived. Also, bump
gopkg.in/yaml.v2 to [gopkg.in/yaml.v3](https://github.com/go-yaml/yaml).

### CODE REVIEW

**Comment by fmuyassarov** [comment]
cc @klihub

File: go.mod [hunk 1]
```diff
@@ -24,5 +24,6 @@ require (
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/xeipuuv/gojsonpointer v0.0.0-20180127040702-4e3ac2762d5f // indirect
 	github.com/xeipuuv/gojsonreference v0.0.0-20180127040603-bd5ef7bd5415 // indirect
-	gopkg.in/yaml.v2 v2.4.0 // indirect
 )
+
+replace github.com/opencontainers/runtime-tools/util/tap v0.0.0 => ./util/tap
```
[the remark this answers is not in the document]
Comment (fmuyassarov) [reply go.mod]: I forgot to clean it up. I was using it for testing purposes but not needed here indeed. Thanks for the catch.

**Comment by fmuyassarov** [comment]
ping @mrunalp @kolyshkin

**Comment by thaJeztah** [comment]
(sorry for drive by comment) 

Instead of a straight copy/paste of the code, perhaps it would be good to migrate the code including history (see https://github.com/moby/sys/pull/121 for an example)

Perhaps also separating the integration commits from the follow ups (formatting build tags, upgrading yaml dependency)

**Comment by fmuyassarov** [comment]
@thaJeztah sorry for my late response on this, I was on PTO.
I will update my patch as you suggested.

**Comment by thaJeztah** [comment]
@fmuyassarov no worries!

Also feel free to "ping" me if you need help with those instructions; I know it can be a bit of a pain to do (and make sure you use temporary repositories on your local machine), but I think it's good to a) preserve history (which can be relevant), and b) preserve the original commits to accredit the authors of those commits.

**Comment by fmuyassarov** [comment]
> @fmuyassarov no worries!
> 
> Also feel free to "ping" me if you need help with those instructions; I know it can be a bit of a pain to do (and make sure you use temporary repositories on your local machine), but I think it's good to a) preserve history (which can be relevant), and b) preserve the original commits to accredit the authors of those commits.

+1

**Pull request closed without merging**